src/qiskit_qasm3_import/expression.py

```diff
@@ -62,43 +76,60 @@ class ValueResolver(QASMVisitor):
     def __init__(self, symbols: Mapping[str, Symbol]):
         self.symbols = symbols
 
-    def resolve(self, node: ast.Expression) -> Tuple[Any, types.Type]:
+    def resolve(self, node: ast.Expression, context=None) -> Tuple[Any, types.Type]:
```

Comment:
We should either store all the relevant parts of the state in the `ValueResolver` instance, or we should pass them all (including the symbol table) through the `context` parameter.  I don't mind particularly which we choose, but let's be consistent.  If we pass the context as an argument, let's also return it in the tuple in `resolve` at the end, so we're free to mutate the context in whatever manner we like, including replacing the whole thing.

Minor: the entry point `resolve` probably shouldn't accept `None` as an input to `context` here, since it's now required.

Reply:
I'm pretty sure the code has been modified to satisfy these items. I'm resolving this to reduce visual clutter.